### PR TITLE
[GPU] Add Softmax/Eltwise Dropout attr support

### PR DESCRIPTION
**Pull request by h-sadia**
This PR is part 2 of adding dropout support for SDPA. This PR covers support for simple softmax kernel and ref eltwise kernel
Part 1 was here: https://github.com/uxlfoundation/oneDNN/pull/4494

### CODE REVIEW

**Comment by h-sadia** [comment]
make test
set test_scope=NIGHTLY
disable test_device_cpu
disable benchdnn_all
enable benchdnn_softmax
enable arch_gpu_xe-hpc
enable arch_gpu_xe-hpg-atsm
enable arch_gpu_xe-hpg-dg2
enable arch_gpu_xe-lp
enable arch_gpu_xe-lpg
enable arch_gpu_xe-lpg+
enable arch_gpu_xe2-hpg-bmg
enable arch_gpu_xe2-lpg
enable arch_gpu_xe3-lpg

**Comment by h-sadia** [comment]
make test
set test_scope=NIGHTLY
disable test_device_cpu
disable benchdnn_all
enable benchdnn_softmax
enable benchdnn_eltwise
enable arch_gpu_xe-hpc
enable arch_gpu_xe-hpg-atsm
enable arch_gpu_xe-hpg-dg2
enable arch_gpu_xe-lp
enable arch_gpu_xe-lpg
enable arch_gpu_xe-lpg+
enable arch_gpu_xe2-hpg-bmg
enable arch_gpu_xe2-lpg
enable arch_gpu_xe3-lpg

**Comment by h-sadia** [comment]
make test
set test_scope=NIGHTLY
disable test_device_cpu
disable benchdnn_all
enable benchdnn_softmax
enable benchdnn_eltwise
enable arch_gpu_xe-hpc

**Review by umar456 — APPROVED** [review]

**Comment by h-sadia** [comment]
make test
set test_scope=NIGHTLY
disable test_device_cpu
disable benchdnn_all
enable benchdnn_softmax
enable benchdnn_eltwise
enable benchdnn_matmul

File: src/gpu/intel/matmul/ref.hpp [hunk 1]
```diff
@@ -247,9 +247,12 @@ struct ref_t : public primitive_t {
     status_t init(impl::engine_t *engine) override {
         compute::kernel_ctx_t kernel_ctx;
 
+        bool with_seed_s64
```
Comment (kealan-barbieri) [inline src/gpu/intel/matmul/ref.hpp:250]: what is the status of non-s64 seed? it is still supported via API but not through benchdnn?

Comment (h-sadia) [reply src/gpu/intel/matmul/ref.hpp:250]: Yes, thats true. the seed was previously s32 in the API.

Comment (kealan-barbieri) [reply src/gpu/intel/matmul/ref.hpp:250]: Seems like something that should be enabled through benchdnn so we can test full range of supported features.

Comment (dzarukin) [reply src/gpu/intel/matmul/ref.hpp:250]: It was a deliberate choice not to test s32 in favor of s64. It's highly unlikely anybody will use this functionality besides PyTorch which has a requirement for int64 seed.

File: src/gpu/intel/include/dropout.h [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+uint get_dropout_threshold(float p) {
+    if (p >= 1.f) return 0xFFFFFFFFu;
+    char exponent = 126 - ((as_uint(p) >> 23) & 0x7F);
+    if ((p <= 0.f) || (exponent > 31)) return 0u;
+    uint mantissa = (as_uint(p) << 8) | 0x80000000u;
+    if (!exponent) return (convert_ulong(mantissa) * 0xFFFFFFFFuL) >> 32;
+    return ((convert_ulong(mantissa >> exponent) * 0xFFFFFFFFuL) >> 32)
+            + !!(mantissa & ((1u << exponent) - 1u));
+}
+#endif
```
Comment (dzarukin) [inline src/gpu/intel/include/dropout.h]: Maybe add to `philox.h` instead? It has `stochastic_round_fwd` already, thus, don't see why `get_dropout_threshold` can't be there as well, surrounded with `WITH_DROPOUT` macro standardizing it across all ocl implementations.

Comment (h-sadia) [reply src/gpu/intel/include/dropout.h]: Yeah, I didn't want it get mixed up with philox code, therefore I created it separately. But maybe I can rename the philox.h since its misleading, or just moved stochastic_round_fwd to it, and keep philox intact.

File: src/gpu/intel/matmul/ref.hpp (hunk 1 above)
Comment (h-sadia) [reply src/gpu/intel/matmul/ref.hpp:250]: keeping it as a keep sake for a while @kealan-barbieri  ;)
